hack/test-end-to-end.sh

```diff
@@ -196,6 +196,11 @@ wait_for_build_start "test"
 wait_for_build "test"
 wait_for_app "test"
 
+echo "[INFO] Starting build from ${STI_CONFIG_FILE} with non-existing commit..."
+set +e
+oc start-build test --commit=fffffff --wait && echo "The build was supposed to fail, but it succeeded." && exit 1
+set -e
```

Comment:
@bparees not sure we want to add one as it will make the end-to-end longer (+~20s)... 
I think the test where we actually wait for build to fail is enough... I will add extended test for cancel anyway, so I can add one extra to verify the wait return 0 when build succeed.